architecture/2024-11-ucp-component-threat-model.md

```diff
@@ -0,0 +1,276 @@
+
+**Impact:** An authorized user of the Kubernetes cluster with permission to create a resource can execute any application definition in any environment registered with Radius.
+
+**Mitigations:**
+
+1. Operators should limit direct access to the Radius API using Kubernetes RBAC.
+2. We should revisit the threat model and provide a more robust set of authorization controls when granular authorization policies are added to Radius.
+
+**Status:** These mitigations are partial and require configuration by the operator. We will revisit and improve this area in the future.
+
+#### Threat: Lack of Role-Based Access Control (RBAC) and Unauthorized Traffic
+
+**Description:** As mentioned above, as of November 15, 2024, the UCP does not implement RBAC, and communication between the UCP and resource providers is unauthenticated.
+
+**Impact:** Increased risk of unauthorized access and actions, making it easier for attackers to interact with resource providers or manipulate user resources without proper authorization.
+
+**Mitigations:**
+
+1. **Implement RBAC within the UCP:** An authentication and authorization mechanism that verifies the identity of clients and enforces access policies must be developed.
+2. **Secure Communication Between UCP and Other Components:** mTLS should be enabled.
+3. **Network Policies and Firewall Rules:** Application of Kubernetes Network Policies to control traffic flow to and from the UCP.
+
+**Status:** Other than the second mitigation (secure communication) listed above, none of the mitigations is currently active.
+
+#### Threat: Use of Weak Cryptographic Hashing Algorithm (SHA-1)
+
+**Description:** As mentioned above, the UCP uses SHA-1 for hashing resource IDs and generating ETags.
+
+**Impact:** Although used for non-security purposes, SHA-1 has known vulnerabilities. If misused in security contexts, it could lead to hash collisions and potential security weaknesses. Also, if we want to have a stronger cryptographic hashing algorithm for hashing resource IDs and generating ETags, we should use another one.
+
+**Mitigations:**
+
+1. **Replace SHA-1 with a Stronger Algorithm.**
+
+**Status:** The mitigation is not active as of now. An action item is created to update the crypto algorithm used in hashing resource IDs and generating ETags as well as in other components of Radius.
+
+## Open Questions
+
+## Action Items
+
+1. Use a hashing algorithm other than SHA-1 while computing the hash of resource IDs and generating ETags.
+2. Check if RBAC with Least Privilege is configured for every component to ensure that each component has only the permissions it needs to function. Make changes to the necessary components if required.
+3. Define and implement necessary Network Policies to ensure that communication is accepted only from expected and authorized components. Regularly review and update these policies to maintain security.
```

Comment:
Let's discuss this

Reply:
Action Item: Secure communication between UCP and Resource Providers. Some form of Authentication and Network Policies and Documentation. mTLS is a form of authentication.